### ISSUE TITLE
Produced class with generics and @Structural is not compilable

### ISSUE THREAD

**Issue by pogo19** [immutables/immutables#354]
I have a class that has generics and Structural annotation. Immutables result (ImmutableEvent) is not compilable.

Anyway, thanks for wonderful project.

```java
@Immutable
@Gson.TypeAdapters
@Structural
public abstract class Event<K, T> {

	@Parameter
	public abstract String getUuid();

	@Parameter
	public abstract long getTimestamp();

	@Parameter
	public abstract int getSchemaVersion();

	@Parameter
	public abstract EventType getType();

	@Parameter
	public abstract String getOriginator();

	@SuppressWarnings("rawtypes")
	@Parameter
	public abstract Class getPayloadIdClass();

	@Parameter
	@Nullable
	public abstract K getPayloadId();

	@SuppressWarnings("rawtypes")
	@Parameter
	public abstract Class getPayloadClass();

	@Parameter
	@Nullable
	public abstract T getPayload();

}
```

**Comment by elucash** [comment]
Most probably `Structural` functionality was not properly updated to use newly-introduced generics. This better to be fixed, thank you for the issue report!

**Comment by elucash** [comment]
Released in 2.2.5. Thank you!

**Comment by pogo19** [comment]
Thanks, no compile problems now.
